Engine/Engine/Core/Application/EngineApplication.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef PATHFINDER_ENGINEAPPLICATION_HPP
+#define PATHFINDER_ENGINEAPPLICATION_HPP
+
+#include <vector>
+#include <memory>
+#include "ApplicationLayer.hpp"
+#include "ScenesLayer.hpp"
+
+#include <SFML/Graphics/RenderWindow.hpp>
+
+namespace Engine
+{
+    class EngineApplication
+    {
+    public:
+        EngineApplication();
+        ~EngineApplication();
+
+        void Run();
+        void Stop();
+
+        void PushLayer(ApplicationLayer* layer);
+        void RemoveLayer(ApplicationLayer* layer);
+
+        inline ScenesLayer& GetScenesLayer() { return m_scenesLayer; }
+
+    private:
+        sf::RenderWindow m_window;
+
+        bool m_running = true;
+        Timestep m_LastFrameTime = 0;
+
+        // TODO : use a custom stack where we can iterate throw
+        std::vector<ApplicationLayer*> m_layers { nullptr };
+        ScenesLayer m_scenesLayer { m_window };
```

Comment:
Je ne vois pas trop l'intérêt de le stocker 2 fois ce layer?

Reply:
`m_layers` sert à itéré sur tous les layers mais `m_scenesLayer` c'est l'instance du ScenesLayer qui est présent dans tout les cas (d'ailleurs le { nullptr } de m_layers est une erreur je l'ai enlevé)
Mais je pense bouger l'instance du ScenesLayer dans une autre class qui gérerait l'instanciation / des-instanciation des scènes